lib/trino-memory-context/src/main/java/io/trino/memory/context/CoarseGrainLocalMemoryContext.java

```diff
@@ -14,32 +14,40 @@
 
 package io.trino.memory.context;
 
+import com.google.common.annotations.VisibleForTesting;
 import com.google.common.util.concurrent.Futures;
 import com.google.common.util.concurrent.ListenableFuture;
 
 import static com.google.common.base.Preconditions.checkArgument;
 import static java.util.Objects.requireNonNull;
 
-public class ThresholdLocalMemoryContext
+/**
+ *  This class prevents contention in the memory tracking system by coarsening the granularity of memory tracking.
+ *  In this paradigm, the most small incremental increases at the byte granularity will not actually result
+ *  in a different coarse granularity value, so no reporting into memory tracking is necessary
+ */
+public class CoarseGrainLocalMemoryContext
```

Comment:
All `LocalMemoryContext` implementations are thread safe. I think this one should be too cc @losipiuk